src/capi/azext_capi/custom.py

```diff
@@ -209,30 +209,44 @@ def create_new_management_cluster(cmd, cluster_name=None, resource_group_name=No
     return True
 
 
+def find_resource_group_name_of_aks_cluster(cluster_name):
+    jmespath_query = "[].{name:name, group:resourceGroup}"
+    command = ["az", "aks", "list", "--query", jmespath_query]
+    result = run_shell_command(command)
+    result = json.loads(result)
+    result = [item for item in result if item['name'] == cluster_name]
+    if result:
+        return result[0]["group"]
+    return None
+
+
 def delete_management_cluster(cmd, yes=False):  # pylint: disable=unused-argument
     exit_if_no_management_cluster()
-    msg = 'Do you want to delete Cluster API components from the current cluster?'
+    cluster_name = kubectl_helpers.find_cluster_in_current_context()
+
+    msg = f"Do you want to delete {cluster_name} cluster?"
+
+    cluster_resource_group = None
+    is_kind_cluster = has_kind_prefix(cluster_name)
+    if not is_kind_cluster:
+        cluster_resource_group = find_resource_group_name_of_aks_cluster(cluster_name)
+        if not cluster_resource_group:
+            prompt = f"Please enter resource group name of {cluster_name} AKS cluster: "
+            cluster_resource_group = get_user_prompt_or_default(prompt, cluster_name, skip_prompt=yes)
+            msg = f"Do you want to delete {cluster_name} cluster and resource group {cluster_resource_group}?"
+
+    pre_workload_warning = f"""\
```

Comment:
Should we do a `kubectl get clusters -A` here so the user knows more specifically what might be orphaned?

```
Please note that cluster "my-cluster" is managing workload clusters:
NAMESPACE   NAME            PHASE         AGE     VERSION
default     default-20520   Provisioned   22m11s
default     default-32106   Provisioned   2m28s
default     default-70206   Provisioning  45s

Deleting the management cluster will leave these workload clusters in their
current configuration, without a way to manage them.
```

Then we could skip the confirmation if it isn't managing any workload clusters.

Reply:
(This could also be an improvement in a later PR. This change basically looks good to me otherwise.)

Reply:
I created #146 to follow up on this idea.